deepchem/data/datasets.py

```diff
@@ -2440,6 +2519,28 @@ def iterate(dataset, batch_size, epochs, deterministic, pad_batches):
 
     return iterate(self, batch_size, epochs, deterministic, pad_batches)
 
+  def __getitem__(self,
+                  index: Union[int, slice]) -> Union[Tuple, Iterator[Batch]]:
+    """Implements indexing for this object."""
+
+    def get_image(array, index):
+      if isinstance(array, np.ndarray):
+        return array[index]
+      return dc.data.ImageLoader.load_img([array[index]])[0]
```

Comment:
I seem this function is used many times in ImageDataset. So, I think it is better to make the method in ImageDataset class directly and replace all `get_image` to `self._get_image`.


```python
  def __getitem__(self,
                  index: Union[int, slice]) -> Union[Batch, Iterator[Batch]]:
    """Implements indexing for this object."""

    if isinstance(index, int):
      if index < 0 or index >= len(self):
        raise IndexError("Index %d is out of bounds" % index)
      return (self._get_image(self._X, index), self._get_image(self._y, index),
              self._w[index], self._ids[index])
    else:
       ......

  def _get_image(self, array: Union[np.ndarray, List[str]],
                 index: int) -> np.ndarray:
    """Method for loading an image

    Parameters
    ----------
    array: Union[np.ndarray, List[str]]
      A numpy array which contains images or List of image filenames
    index: int
      Index you want to get the image

    Returns
    -------
    np.ndarray
      Loaded image
    """
    if isinstance(array, np.ndarray):
      return array[index]
    return ImageLoader.load_img([array[index]])[0]
```